src/Draggable/Sensors/DragSensor/tests/DragSensor.test.js

```diff
@@ -0,0 +1,514 @@
+        expect(container.addEventListener.mock.calls.length).toBe(0);
+      });
+
+      dragSensor.attach();
+
+      // After calling attach, 5 events should have been added
+      containers.forEach((container) => {
+        expect(container.addEventListener.mock.calls.length).toBe(5);
+      });
+
+      // And the events should be correct
+      containers.forEach((container) => {
+        expect(container.addEventListener.mock.calls[0])
+          .toMatchObject(['mousedown', dragSensor._onMouseDown, true]);
+
+        expect(container.addEventListener.mock.calls[1])
+          .toMatchObject(['dragstart', dragSensor._onDragStart, false]);
+
+        expect(container.addEventListener.mock.calls[2])
+          .toMatchObject(['dragover', dragSensor._onDragOver, false]);
+
+        expect(container.addEventListener.mock.calls[3])
+          .toMatchObject(['dragend', dragSensor._onDragEnd, false]);
+
+        expect(container.addEventListener.mock.calls[4])
+          .toMatchObject(['drop', dragSensor._onDragDrop, false]);
+      });
+    });
+
+    test('should add mouseup event listener to document', (done) => {
+      const dragSensor = new DragSensor();
+
+      const originalAddEventListener = document.addEventListener;
+
+      document.addEventListener = jest.fn();
+
+      dragSensor.attach();
+
+      expect(document.addEventListener.mock.calls.length).toBe(1);
+
+      document.addEventListener = originalAddEventListener;
+
+      done();
+    });
+  });
+
+  describe('#detach', () => {
+    let container = null;
+
+    beforeEach(() => {
+      container = document.createElement('div');
+      container.removeEventListener = jest.fn();
+      document.removeEventListener = jest.fn();
+    });
+
+    afterEach(() => {
+      container = null;
+    });
+
+    test('should remove event listeners', () => {
+      const dragSensor = new DragSensor([container]);
+
+      dragSensor.detach();
+
+      expect(container.removeEventListener.mock.calls.length).toBe(5);
+
+      expect(container.removeEventListener.mock.calls[0])
+        .toMatchObject(['mousedown', dragSensor._onMouseDown, true]);
+
+      expect(container.removeEventListener.mock.calls[1])
+        .toMatchObject(['dragstart', dragSensor._onDragStart, false]);
+
+      expect(container.removeEventListener.mock.calls[2])
+        .toMatchObject(['dragover', dragSensor._onDragOver, false]);
+
+      expect(container.removeEventListener.mock.calls[3])
+        .toMatchObject(['dragend', dragSensor._onDragEnd, false]);
+
+      expect(container.removeEventListener.mock.calls[4])
+        .toMatchObject(['drop', dragSensor._onDragDrop, false]);
+    });
+  });
+
+  describe('#_onDragStart', () => {
+    let sandbox;
+    let dragSensor;
+
+    beforeEach(() => {
+      listenToSensorEvents();
+      sandbox = createSandbox(sampleMarkup);
+      const container = sandbox.querySelectorAll('ul');
+      dragSensor = new DragSensor(container, {
+        delay: 0,
```

Comment:
damn, we should make that a default for Sensors. We'll do in a follow-up PR

Reply:
👍